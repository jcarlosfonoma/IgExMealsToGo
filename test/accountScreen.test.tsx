import React from "react"
import { create } from "react-test-renderer"
import { AccountScreen } from "../app/features/account/screens/account.screen"
import { ThemeProvider } from "styled-components/native"
import { theme } from "../app/infrastructure/theme"

const navigation = {
  navigate: jest.fn(),
}

const tree = create(
  <ThemeProvider theme={theme}>
    <AccountScreen navigation={navigation} />
  </ThemeProvider>,
)

jest.mock("lottie-react-native")
jest.requireActual("lottie-react-native")

describe("Account screen tests", () => {
  test("should match snapshot", () => {
    expect(tree).toMatchSnapshot()
  })
})
