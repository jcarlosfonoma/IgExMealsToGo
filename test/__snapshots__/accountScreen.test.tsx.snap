// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account screen tests should match snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "height": "40%",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "position": "absolute",
          "top": 30,
          "width": "100%",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "flexWrap": "wrap",
          "fontFamily": "Oswald_400Regular",
          "fontSize": 30,
          "fontWeight": "400",
          "marginBottom": 0,
          "marginTop": 0,
        },
      ]
    }
    variant="body"
  >
    Meals To Go
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.7)",
          "marginTop": 8,
          "paddingBottom": 32,
          "paddingLeft": 32,
          "paddingRight": 32,
          "paddingTop": 32,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#2182BD",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "minWidth": 64,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
              }
            }
          >
            <Text
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontSize": 16,
                  },
                ]
              }
            >
              □
            </Text>
          </View>
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 16,
          },
        ]
      }
      variant="marginTop:16px"
    >
      <View
        style={
          Object {
            "backgroundColor": "#2182BD",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "minWidth": 64,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 12,
                  "marginRight": -4,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontSize": 16,
                    },
                  ]
                }
              >
                □
              </Text>
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Register
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
